### PR TITLE
New: convert Heading to TS

### PR DESCRIPTION
**Pull request by aabenoja**

### CODE REVIEW

**Review by MarioLozanoM — APPROVED** [review]

File: packages/es-components/src/components/containers/heading/Heading.tsx [hunk 1]
```diff
@@ -1,23 +1,48 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import styled, { css } from 'styled-components';
+import styled, {
+  DefaultTheme,
+  ThemedStyledProps,
+  css
+} from 'styled-components';
+import { HeadingLevel } from '../../../../../../shared/types/src/ESTheme';
```
Comment (stevematney) [inline packages/es-components/src/components/containers/heading/Heading.tsx:8]: We have an alias for this.

```suggestion
import { HeadingLevel } from 'es-components-shared-types';
```

File: packages/es-components/src/components/containers/heading/Heading.tsx [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
 const knockoutStyles = css`
   background-color: ${({ theme }) => theme.colors.primary};
   color: white;
   padding: 20px 15px;
 `;
 
-const getAdjustedProps = func => ({ size, level = 1, ...props }) =>
-  func({
-    ...props,
+export type HeadingProps = Omit<JSX.IntrinsicElements['h1'], 'ref'> & {
```
Comment (stevematney) [inline packages/es-components/src/components/containers/heading/Heading.tsx:16]: There is a global helper we added for this.
```suggestion
export type HeadingProps = JSXElementProps<'h1'> & {
```

File: packages/es-components/src/components/containers/modal/ModalHeader.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import styled from 'styled-components';
 import OrigDismissButton from '../../controls/DismissButton';
 import { ModalContext } from './ModalContext';
 import Heading from '../heading/Heading';
+import { HeadingLevel } from '../../../../../../shared/types/src/ESTheme';
```
Comment (stevematney) [inline packages/es-components/src/components/containers/modal/ModalHeader.tsx:8]:
```suggestion
import { HeadingLevel } from 'es-components-shared-types';
```

**Review by stevematney — CHANGES_REQUESTED** [review]
Just a couple of really minor things, but this looks good overall!

**Pull request merged**